lib/core/src/lib/datatable/components/date-cell/date-cell.component.ts

```diff
@@ -15,67 +15,55 @@
-            </span>
-        </ng-container>
-        <ng-template #standard_date>
-            <span
-                class="adf-datatable-cell-value"
-                title="{{ tooltip | adfLocalizedDate: tooltipDateFormat }}">
-                {{ value$ | async | adfLocalizedDate: format }}
-            </span>
-        </ng-template>
-    `,
+    templateUrl: './date-cell.component.html',
     encapsulation: ViewEncapsulation.None,
-    host: { class: 'adf-date-cell adf-datatable-content-cell' }
+    host: { class: 'adf-date-cell adf-datatable-content-cell' },
+    changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class DateCellComponent extends DataTableCellComponent {
+export class DateCellComponent extends DataTableCellComponent implements OnInit {
 
-    static DATE_FORMAT = 'medium';
+    @Input()
+    dateConfig: DateConfig;
 
-    currentLocale: string;
-    dateFormat: string;
-    tooltipDateFormat: string;
+    config: DateConfig = {};
 
-    get format(): string {
-        if (this.column) {
-            return this.column.format || this.dateFormat;
-        }
-        return this.dateFormat;
+    readonly defaultDateConfig: DateConfig = {
+        format: 'medium',
+        tooltipFormat: 'medium',
+        locale: undefined
+    };
+
+    constructor(private readonly appConfig: AppConfigService) {
+        super();
     }
 
-    constructor(
-        userPreferenceService: UserPreferencesService,
-        @Optional() dataTableService: DataTableService,
-        appConfig: AppConfigService
-    ) {
-        super(dataTableService);
+    ngOnInit(): void {
+        super.ngOnInit();
+        this.setDefaultConfig();
+    }
 
-        this.dateFormat = appConfig.get('dateValues.defaultDateFormat', DateCellComponent.DATE_FORMAT);
-        this.tooltipDateFormat = appConfig.get('dateValues.defaultTooltipDateFormat', DateCellComponent.DATE_FORMAT);
-        if (userPreferenceService) {
-            userPreferenceService
-                .select(UserPreferenceValues.Locale)
-                .pipe(takeUntil(this.onDestroy$))
-                .subscribe(locale => this.currentLocale = locale);
+    private setDefaultConfig(): void {
+        if (this.dateConfig) {
+            this.config.format = this.dateConfig?.format ?? (this.column?.format || this.getAppConfigPropertyValue('dateValues.defaultDateFormat', this.defaultDateConfig.format));
+            this.config.tooltipFormat = this.dateConfig?.tooltipFormat || this.getAppConfigPropertyValue('dateValues.defaultTooltipDateFormat', this.defaultDateConfig.tooltipFormat);
+            this.config.locale = this.dateConfig?.locale || this.getAppConfigPropertyValue('dateValues.defaultLocale', this.defaultDateConfig.locale);
+        } else {
+            this.config = this.defaultDateConfig;
         }
     }
+
+    private getAppConfigPropertyValue(key: string, defaultValue: string): string {
+        return this.appConfig.get(key, defaultValue);
+    }
```

Comment:
```suggestion
    private setDefaultConfig(): void {
        if(!this.dateConfig){
            this.config = this.defaultDateConfig;
            return;
        }
        this.config.format = this.dateConfig?.format ?? (this.column?.format || this.getAppConfigPropertyValue('dateValues.defaultDateFormat', 'format'));
        this.config.tooltipFormat = this.dateConfig?.tooltipFormat || this.getAppConfigPropertyValue('dateValues.defaultTooltipDateFormat', 'tooltipFormat');
        this.config.locale = this.dateConfig?.locale || this.getAppConfigPropertyValue('dateValues.defaultLocale', 'locale');
    }

    private getAppConfigPropertyValue(key: string, defaultDateConfigKey: keyof DateConfig): string {
        return this.appConfig.get(key, this.defaultDateConfig[defaultDateConfigKey]);
    }
```
WDYT about that implementation? not sure which is better but that's just another point of view. Maybe this is more understandable?

Reply:
Btw in this example is nice to avoid the else{} block and just handle 'smaller' stuff at the beginning and get use of `return;` I think